pilot/pkg/features/pilot.go

```diff
@@ -403,6 +403,14 @@ var (
 		"If enabled, pilot will authorize XDS clients, to ensure they are acting only as namespaces they have permissions for.",
 	).Get()
 
+	// TODO: Move this to proper API.
+	TrustedGatewayCIDR = env.RegisterStringVar(
+		"TRUSTED_GATEWAY_CIDR",
+		"",
+		"If set, any connections from gateway to Istiod with this CIDR range are treated as trusted for using authenication mechanisms like XFCC."+
+			" This can only be used when the network where Istiod and the authenticating gateways are running is a trusted/secure network",
+	).Get()
```

Comment:
cidr does not reflect the xfcc

Reply:
Can you elobrate more? We are saying here that XFCC is used only when the connection is from trusted CIDR.

Reply:
cidr has no direct relationship with xfcc headers

Reply:
It has no direct relationship - but please comments if xfcc authenticator has to be used, the connection address should be in trusted CIDR. Otherwise we can not trust XFCC header.

Reply:
That was the whole discussion we had in this PR. Even if user trusts it, how do you guarantee that xfcc is not spoofed?

Reply:
If you config it, you trust this way, just like k8s request header authenticator. It leaves the decision to the user. 
In another word, this is a  weaker authentication compared to others

Reply:
> If you config it, you trust this way, just like k8s request header authenticator. It leaves the decision to the user.

request header authenticator does not leave it to user https://kubernetes.io/docs/reference/access-authn-authz/authentication/#authenticating-proxy

_In order to prevent header spoofing, the authenticating proxy is required to present a valid client certificate to the API server for validation against the specified CA before the request headers are checked. WARNING: do not reuse a CA that is used in a different context unless you understand the risks and the mechanisms to protect the CA's usage._

It is almost similar here

Reply:
I donot think cidr make much sense as in istio or k8s scenario, istiod within a cluster can not be accessed from outer clients if not exposed Loadbalancer type service or Ingress.

And also many cni support networkPolicy, which can limit the access from underlying network level.